### ISSUE TITLE
magma 2.x.x support

### ISSUE THREAD

**Issue by cpviolator** [lattice/quda#547]
I'm compiling on Tesla P100 and I wanted to use magma-2.2.0 to see if there was any significant increase in multigrid over 1.7.0. Compiling for 2.x.x requires a few function name changes, which is no big deal. However, when calling the functions

```c++
magma_int_t nb = magma_get_cgeqrf_nb(m);
magma_int_t nb = magma_get_zgeqrf_nb(m);
```
used for EigCG(?), one must pass a second argument to the functions. As far as I can tell, the original argument is the number of columns in the QR decomposition, and the new second argument is, I assume, the number of rows. Magma makes decisions based on these two inputs, based on which is the greater. Compilation is of course successful when making the adjustment,
```c++
magma_int_t nb = magma_get_cgeqrf_nb(m,m);
magma_int_t nb = magma_get_zgeqrf_nb(m,m);
```
but I imagine there is a more useful parameter to pass than a repeated `m`. I'm not entirely up-to-speed on EigCG, so I don't know what that second parameter should be. If somebody can shed some light here, I can create a hotfix where we can compile with magma 2.x.x using a boolean in cmake and some pre-processor flags in `blas_magma.cu` with function calls relevant for the magma version.

@maddyscientist @alexstrel any ideas?

**Comment by alexstrel** [comment]
@cpviolator , fyi, I updated eigCG solver to be used with eigen library only, in the feature/fgmresdr, though it needs to be tested. I know Kate wanted to remove magma call from multigrid as well. That is, magma contribution will be minimized in the future releases.  Could you try to update blas_magma.cu with https://github.com/lattice/quda/blob/feature/staggered-multigrid/lib/blas_magma.cu (note lines,e.g.,  425-429) ?

**Comment by alexstrel** [comment]
Are you interesting in multigrid or eigCG performance?

**Comment by cpviolator** [comment]
Those lines (452-429 among others) are exactly the type of updates I wrote. I suspected that the second argument was related to the number of eigenvectors, and I see that in `blas_magma.cu` one uses the `htsize` variable.

My strategy is to include a boolean variable in the CMake build `QUDA_MAGMA_VERSION_GE20` (because I don't know how to automatically scrape the magma version) so that if the user knows they are using magma-2.x.x they simply switch that boolean `ON` and the correct functions are called. I'll test this with some data to ensure everything is correct then push the changes.

For the moment I'm more interested in multigrid, but if I get around to writing an overlap dslash then I'll also want efficient left- right-eigenvector calculation, and possibly SVD.
